### PR TITLE
new: Add support for Parent/Child account switching

### PR DESCRIPTION
**Pull request by lgarber-akamai**
## 📝 Description

This change adds preemptive support for the Parent/Child Account Switching feature, including the following changes:
- New endpoint methods:
    - client.ListChildAccounts(...) 
    - client.GetChildAccount(...)
    - client.CreateChildAccountToken(...)
- Updated structs:
    - linodego.User now contains a `UserType` field

In addition to the above changes, this change brings the `linodego.Account` struct up to parity with the current API response.
 
This change also makes the following testing/helper changes:

- The `test` package now has a `unit` directory for storing mock tests
- The `test` package now imports `testify/require` to make tests more concise

Depends on https://github.com/linode/linodego/pull/458

## ✔️ How to Test

The following test steps assume you have pulled down this PR locally and are using the token and URL for a parent account in dev.

### Unit Testing

```bash
make testunit
```

### E2E Testing

```bash
make ARGS="-run TestUser" fixtures
make TEST_TAGS="parent_child" ARGS="-run TestAccount" fixtures
```

### Manual Testing

1. In a manual testing environment (e.g. dx-devenv), run the following script:

```go
package main

import (
	"context"
	"fmt"
	"log"
	"os"

	"github.com/linode/linodego"
	"github.com/sanity-io/litter"
)

func main() {
	ctx := context.Background()

	client := linodego.NewClient(nil)
	client.SetToken(os.Getenv("LINODE_TOKEN"))

	childAccounts, err := client.ListChildAccounts(ctx, nil)
	if err != nil {
		log.Fatal(err)
	}

	childAccount, err := client.GetChildAccount(ctx, childAccounts[0].EUUID)
	if err != nil {
		log.Fatal(err)
	}

	token, err := client.CreateChildAccountToken(ctx, childAccount.EUUID)
	if err != nil {
		log.Fatal(err)
	}

	fmt.Println("Child Accounts:")
	litter.Dump(childAccounts)

	fmt.Println("\nChild Account:")
	litter.Dump(childAccount)

	fmt.Println("\nChild Account Token:")
	litter.Dump(token)
}
```

2. Ensure the output looks similar to the following:

```
Child Accounts:
[]linodego.Account{
  linodego.Account{
    FirstName: "DX-Child",
    ...
  },
}

Child Account:
&linodego.Account{
  FirstName: "DX-Child",
  ...
}

Child Account Token:
&linodego.Token{
  ID: ...,
  Token: ...,
}
```

### CODE REVIEW

File: account_child.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+type ChildAccountToken = Token
+
+// ChildAccountsPagedResponse represents a Linode API response
+// for listing child accounts under the current account.
+type ChildAccountsPagedResponse struct {
+	*PageOptions
+	Data []ChildAccount `json:"data"`
+}
+
+// endpoint gets the endpoint URL for Instance
```
Comment (yec-akamai) [inline account_child.go:27]: nit: Seems like a typo in the comment

File: account.go [hunk 2]
```diff
@@ -20,6 +26,29 @@ type Account struct {
 	Phone             string      `json:"phone"`
 	CreditCard        *CreditCard `json:"credit_card"`
 	EUUID             string      `json:"euuid"`
+	BillingSource     string      `json:"billing_source"`
+	Capabilities      []string    `json:"capabilities"`
+	ActiveSince       *time.Time  `json:"active_since"`
```
Comment (yec-akamai) [inline account.go:31]: Should we do
```suggestion
	ActiveSince       *time.Time  `json:"-"`
```

**Review by yec-akamai — APPROVED** [review]
Existing tests are passing 👍

File: request_helpers.go [hunk 3]
```diff
@@ -118,16 +118,22 @@ func doPOSTRequest[T, O any](
 	ctx context.Context,
 	client *Client,
 	endpoint string,
-	options O,
+	options ...O,
```
Comment (lgarber-akamai) [inline request_helpers.go:121]: Necessary for `nil` request bodies

**Comment by lgarber-akamai** [comment]
Going to convert this back to a draft while I work to break some of the smaller changes into separate PRs

**Comment by zliang-akamai** [comment]
`TEST_TAGS` wasn't added into the Makefile?
`make TEST_TAGS="parent_child" ARGS="-run TestAccount" fixtures` doesn't run any test case on my end..

**Review by zliang-akamai — APPROVED** [review]
Everything else tested and looks good!

**Comment by lgarber-akamai** [comment]
@zliang-akamai Oops, looks like it was accidentally dropped when I was fixing some merge conflicts. I can fix it real quick 👍

**Review by zliang-akamai — APPROVED** [review]
Nice work!

**Review by yec-akamai — APPROVED** [review]
Nice work! Change looks good to me.

Unit tests and user tests passed locally. I still don't have the 1pass vault set up for my account, so can't run through the parent/child integration tests on my end.

**Pull request merged**